File: smelly/duplicate-assert/java-junit/JavadocTagInfoTest_testSerial.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class JavadocTagInfoTest {

    @Test
    public void testSerial() {
        DetailAstImpl ast = new DetailAstImpl();
        DetailAstImpl astParent = new DetailAstImpl();
        astParent.setType(TokenTypes.LiteralCatch);
        var setParent = ast.getClass().getDeclaredMethod("setParent", DetailAstImpl.class);
        setParent.setAccessible(true);
        setParent.invoke(ast, astParent);

        TokenTypes[] validTypes = new TokenTypes[]{TokenTypes.VariableDef};
        for (TokenTypes type : validTypes) {
            ast.setType(type);
            assertTrue(JavadocTagInfo.Serial.isValidOn(ast), "Invalid ast type for current tag: " + ast.getClass());
        }

        astParent.setType(TokenTypes.Slist);
        ast.setType(TokenTypes.VariableDef);
        assertFalse(JavadocTagInfo.Serial.isValidOn(ast), "Should return false when ast type is invalid for current tag");

        ast.setType(TokenTypes.ParameterDef);
        assertFalse(JavadocTagInfo.Serial.isValidOn(ast), "Should return false when ast type is invalid for current tag");
    }
}

File: smelly/duplicate-assert/CSharpXUnit/CSharpXUnit.Tests/Tests/JavadocTagInfoTest_testSerial.cs
using Xunit;

public class JavadocTagInfoTest
{
    [Fact]
    public void TestSerial()
    {
        var ast = new DetailAstImpl();
        var astParent = new DetailAstImpl();
        astParent.SetType(TokenTypes.LiteralCatch);
        var setParent = ast.GetType().GetMethod("SetParent", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
        setParent.Invoke(ast, new object[] { astParent });

        var validTypes = new[] { TokenTypes.VariableDef };
        foreach (var type in validTypes)
        {
            ast.SetType(type);
            Assert.True(JavadocTagInfo.Serial.IsValidOn(ast), $"Invalid ast type for current tag: {ast.GetType()}");
        }

        astParent.SetType(TokenTypes.Slist);
        ast.SetType(TokenTypes.VariableDef);
        Assert.False(JavadocTagInfo.Serial.IsValidOn(ast), "Should return false when ast type is invalid for current tag");

        ast.SetType(TokenTypes.ParameterDef);
        Assert.False(JavadocTagInfo.Serial.IsValidOn(ast), "Should return false when ast type is invalid for current tag");
    }
}
